packages/frontend/app/(dashboard)/course/[cid]/queue/[qid]/components/modals/EditQueueModal.tsx

```diff
@@ -177,6 +222,73 @@ const EditQueueModal: React.FC<EditQueueModalProps> = ({
               message.error(`Failed to change zoom link: ${errorMessage}`)
             })
         : Promise.resolve()
+    const newQueueConfig: QueueConfig = {
+      ...lastSavedQueueConfig.current,
+      minimum_tags: Number(values.minTags),
+      assignment_id: values.assignmentId,
+      // iterate over each task and accumulate them into an object
+      tasks: values.assignmentId
+        ? values.tasks?.reduce((acc, task) => {
+            acc[task.id] = {
+              display_name: task.display_name,
+              short_display_name: task.short_display_name,
+              blocking: task.blocking,
+              color_hex: task.color_hex,
```

Comment:
This is an object, not a string as the backend intends. (But I noticed this while you were working on it so you fixed it actively.)